ext/opentelemetry-ext-requests/src/opentelemetry/ext/requests/__init__.py

```diff
@@ -92,8 +92,9 @@ def instrumented_request(self, method, url, *args, **kwargs):
             span.set_attribute("http.method", method.upper())
             span.set_attribute("http.url", url)
 
-            headers = kwargs.setdefault("headers", {})
+            headers = kwargs.get("headers", {}) or {}
             propagators.inject(type(headers).__setitem__, headers)
+            kwargs["headers"] = headers
```

Comment:
Sorry it took me this long to get back to you, what I meant by my question was could this solve the problem instead of setting the headers attribute:
```suggestion
            headers = kwargs.get("headers", {})
            if headers is not None:
                propagators.inject(type(headers).__setitem__, headers)
```

Reply:
in this case request will not be traced, is that what we want? then it will be implicit

Reply:
agreed that this approach should be taken.

Regarding the need for this PR at all: I guess requests does accept a None as headers. But looking at the API documentation, I don't see None as a valid value to pass in: https://requests.readthedocs.io/en/master/api/#requests.request

Reply:
@codeboten 
I might be missing something here, if the header is `None`, we don't propagate the trace context but we still create the span? Would the trace just end there?

Reply:
That's correct @lzchen, no trace would be propagated here. I guess this is a bit of a strange case, but if as a user I explicitly set headers to `None` when sending a request, I'm not sure I would expect an instrumentation library to add a header regardless of my initial setting.

Reply:
> in this case request will not be traced, is that what we want? then it will be implicit

@HiveTraum to clarify: propagators should not inject anything into headers if no such headers exist. At least, that's the standard that's used for every propagation scheme I know.

I think in this case it's fine, and I would argue more elegant than what's there now. Adding this skips the need to replace the headers entirely in the kwargs on line #97

Reply:
`
@dataclass
class OrgStructureServiceEndpoint:
    url: str
    method: str = 'GET'
    headers: dict = None
    domain: str = None
    params: dict = None

    def send(self):
        url = f"{self.domain or settings.ORGSTRUCT_URL}/{self.url}"
        try:
            response = requests.request(
                method=self.method, url=url, headers=self.headers,
                params=self.params, timeout=settings.REQUEST_TIMEOUT)
        except requests.exceptions.Timeout:
            raise APIException("Не удалось связаться с сервисом Оргструктуры")
`

I have such example which shows implicit scenario when user wants to pass `None` headers implicitly. As user i want to see this request in traces.

I think that if user wants to send request without headers at all - `suppress_instrumentation` must be used and `headers=None`

Reply:
Okay, got it. Yeah I see that case now. Looking at the requests source code, it looks like things are handled in a very similar way (default to an empty dict):

```
    def __init__(self,
            method=None, url=None, headers=None, files=None, data=None,
            params=None, auth=None, cookies=None, hooks=None, json=None):

        # Default empty dicts for dict params.
        data = [] if data is None else data
        files = [] if files is None else files
        headers = {} if headers is None else headers
        params = {} if params is None else params
        hooks = {} if hooks is None else hooks
```
So yes, this make sense. I'll switch to approved. Thanks!

@codeboten so you can see this too ^

Reply:
Thanks for following up on this. Approving